src/services/git/index.ts

```diff
@@ -1,14 +1,26 @@
 import { exec } from 'child_process'
+import mv from 'mv'
+import fs from 'fs'
 
 export class GitService {
-  public static clone = async (repository: string, destination: string): Promise<void> => {
+  public static clone = async (
+    repository: string,
+    useCase: string,
+    destination: string,
+  ): Promise<void> => {
     return new Promise((resolve, reject) => {
-      exec(`git clone ${repository} "${destination}"`, (error) => {
+      exec(`git clone ${repository} "${process.cwd()}/temp-app"`, (error) => {
         if (error) {
           reject(error)
         }
+        mv(`${process.cwd()}/temp-app/use-cases/${useCase}`, `${destination}`, (error) => {
+          if (error) {
+            reject(error)
+          }
+          fs.rmSync(`${process.cwd()}/temp-app`, { recursive: true, force: true })
 
-        resolve()
+          resolve()
+        })
       })
     })
   }
```

Comment:
1. It's better to avoid using business logic (use cases) in such generic service. In this case use case is just a subdirectory, right?
2. We can try to use built-in `fs` method for creating temp directories and use `fs.cp()` instead of `mv` package
3. Avoid using callbacks instead of promises where possible

My (untested) version of this file:
```ts
import os from 'os'
import path from 'path'
import fs from 'fs-extra'

export class GitService {
  public static clone = async (
    repository: string,
    destination: string,
    options?: { subdirectory?: string },
  ): Promise<void> => {
    const tempDir = await fs.mkdtemp(path.join(os.tmpdir(), 'affinidi-cli-'))

    await new Promise<void>((resolve, reject) => {
      exec(`git clone ${repository} "${tempDir}"`, (error) => {
        if (error) {
          reject(error)
        }

        resolve()
      })
    })

    const source = options?.subdirectory
      ? path.join(tempDir, options.subdirectory)
      : tempDir

    await fs.move(source, destination)
    await fs.remove(source).catch(() => {})
  }
}
```

(I'm using `fs-extra` library for `move` and `remove` methods)

Would be used like that:
```
GitService.clone('.../reference-app-certification-and-verification', '...', { subdirectory: `use-cases/health` })
```